### DISCUSSION TITLE
Add UUIDs for datasets / charts / dashboards

### DISCUSSION THREAD

**Discussion by JustGui** [apache/superset#27854] [Ideas]
As explained in the issue https://github.com/apache/superset/issues/15456, the UUIDs are mandatory when trying to create a dataset/chart/dashboard via API calls for the following endpoints:
- /api/v1/chart/import/
- /api/v1/dashboard/
- /api/v1/dashboard/import/
- /api/v1/dataset/import/

Nevertheless, it's not returned by the API on the GET endpoints (tried on v3.1.0, v3.1.1, v4.0.0rc1 & v4.0.0rc2 of Superset)

The only way to know the UUID seems to be when creating the dataset/chart on the endpoint `import` as we push the UUID ourselves.

Two modifications therefore seem necessary:
- Add an endpoint to search by UUIDs the dataset/chart/dashboard
  - When we create via the `import` endpoint as the ID of the dataset/chart/dashboard created is not returned

- Return the UUIDs on the following endpoints:
  - /api/v1/chart/
  - /api/v1/chart/{pk}
  - /api/v1/dashboard/
  - /api/v1/dashboard/{id_or_slug}
  - /api/v1/dashboard/{id_or_slug}/charts
  - /api/v1/dashboard/{id_or_slug}/datasets
  - /api/v1/dataset/
  - /api/v1/dataset/{pk}

Thx and have a nice day!

**Comment by catalingmn** [comment]
that would be great to have for my use cases as well
